react-components/src/analysis/view/AnalysesView.js

```diff
@@ -58,6 +58,20 @@ import analysesCollapseIcon from "../../resources/images/analyses-collapseList.s
 import altAnalysesExpandIcon from "../../resources/images/analyses-expandList.png";
 import altAnalysesCollapseIcon from "../../resources/images/analyses-collapseList.png";
 
+const rangeSelect = (data, selected, deSelect, rangeStart, rangeEnd) => {
+    const newSelectionRange = data.slice(rangeStart, rangeEnd + 1);
+
+    if (deSelect) {
+        // deselect range
+        return selected.filter(
+            (selectedID) => !newSelectionRange.includes(selectedID)
+        );
+    }
+
+    // append range, removing duplicates
+    return [...new Set([...selected, ...newSelectionRange])];
+};
```

Comment:
This `rangeSelect` logic probably didn't have to be pulled out into its own function, but I extracted it in case we wanted something general like this in `ui-lib` for other listings to use. If we do use it outside this view, it could probably use some null checks for `data` and `selected`.